src/Compilers/VisualBasic/Portable/Binding/LookupResult.vb

```diff
@@ -691,6 +691,12 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                     CompactSymbols(lost)
                     Debug.Assert(_symList.Count > 0)
 
+                    ' As a special case, we allow conflicting enum members imported from
+                    ' metadata If the have the same value, and take the first
```

Comment:
The order is indeed deterministic, both in previous compilers and in Roslyn. That is necessary because in the use case of interest for this change, you want reflection to report the new (first) spelling. It is because these members are put out in deterministic order that ToString is able to use the first spelling for a given value. In the cases of interest for this change, the deprecated member is not the first member.